app/server.py

```diff
@@ -22,6 +22,504 @@ def allRecords():
         results.append({'course': document['course'], 'data': document['data'], 'instructor': document['instructor'] })
     return jsonify({ 'count': cursor.count() , 'results': results})
 
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+import hashlib  # Hashing module
+import time # Time module
+import jwt  # JSON Web Token module
+
+# Globals
+db_users = client.userdb
+ISSUER = "reconFIU_server"
+EXP_TIME = 86400 # 1 day
+SECRET_KEY = "RECON_FIU_CEN_4010"  # TODO: Should be Randomized and sent to Client Browser
+
+# Show all documents in userdb
+@app.route('/getallusers')
+def allUsers():
+    cursor = db_users.users.find().limit(100)
+    results = []
+    for document in cursor:
+        results.append({'username': document['username'], 'password': document['password'], 'token': document['token'] })
+    return jsonify({ 'count': cursor.count() , 'results': results})
+
+
+# Find a user in userdb
+@app.route('/finduser/<username>')
+def find_user(username):
+    user = db_users.users.find_one({'username': username})
+    if user:
+        results = []
+        results.append({'username': user['username'], 'password': user['password'], 'token': user['token'] })
```

Comment:
Only thing to be returned should be the user token. Not password or user, user is acceptable tho but it's pointless because the client already has the username